contracts/protocol/libraries/logic/MarketplaceLogic.sol

```diff
@@ -499,10 +497,7 @@ library MarketplaceLogic {
                 bool isNToken = reservesData[underlyingAsset].xTokenAddress ==
                     token;
                 require(isNToken, Errors.ASSET_NOT_LISTED);
-                require(
-                    INToken(token).ownerOf(tokenId) == onBehalfOf,
-                    Errors.INVALID_RECIPIENT
-                );
+                IERC721(token).transferFrom(address(this), onBehalfOf, tokenId);
```

Comment:
use `safeTransferFrom` instead?

Reply:
for NToken it's the same because we didn't implement the real `safeTransferFrom` yet

Reply:
I can switch to that in case  one day we change `safeTransferFrom` to add back the hook